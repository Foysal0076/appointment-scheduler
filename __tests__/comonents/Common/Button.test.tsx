import Button from '@/components/Common/Button'
import { render } from '@testing-library/react'

describe('Button component', () => {
  test('renders without crashing', () => {
    const { container } = render(<Button />)
    expect(container).toBeInTheDocument()
  })

  test('applies default color and size classes if not provided', () => {
    const { container } = render(<Button />)
    const buttonElement = container.querySelector('button')
    expect(buttonElement).toHaveClass('bg-primary-500')
    expect(buttonElement).toHaveClass('h-[2.5rem]')
  })

  test('applies provided color and size classes', () => {
    const { container } = render(<Button color='secondary' size='lg' />)
    const buttonElement = container.querySelector('button')
    expect(buttonElement).toHaveClass('bg-secondary-500')
    expect(buttonElement).toHaveClass('h-[3.125rem]')
  })

  test('disables the button when loading is true', () => {
    const { container } = render(<Button loading />)
    const buttonElement = container.querySelector('button')
    expect(buttonElement).toBeDisabled()
  })

  test('renders spinner when loading is true', () => {
    const { container } = render(<Button loading />)
    const spinnerElement = container.querySelector('.absolute')
    expect(spinnerElement).toBeInTheDocument()
  })

  test('renders children when not loading', () => {
    const { container } = render(<Button>Click me</Button>)
    const childrenElement = container.querySelector('.visible')
    expect(childrenElement).toBeInTheDocument()
    expect(childrenElement).toHaveTextContent('Click me')
  })

  test('matches snapshot', () => {
    const { asFragment } = render(<Button />)
    expect(asFragment()).toMatchSnapshot()
  })
})
