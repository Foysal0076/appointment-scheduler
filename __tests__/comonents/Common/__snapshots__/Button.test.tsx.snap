// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="relative select-none rounded text-white focus:ring-4 disabled:active:transform-none active:scale-[.9] disabled:cursor-not-allowed transition-all duration-300 text-nowrap focus:outline-none font-medium bg-primary-500 disabled:text-primary-200 disabled:bg-primary-50 hover:bg-primary-600 focus:ring-primary-100 focus:outline-primary-200  h-[2.5rem] text-md px-4"
  >
    <div
      class="visible font-Rubik"
    />
  </button>
</DocumentFragment>
`;
